integration-tests/bats/init.bats

```diff
@@ -0,0 +1,142 @@
+#!/usr/bin/env bats
+load $BATS_TEST_DIRNAME/helper/common.bash
+
+setup() {
+    setup_no_dolt_init
+    stash_current_dolt_user
+}
+
+teardown() {
+    restore_stashed_dolt_user
+    assert_feature_version
+    teardown_common
+}
+
+@test "init: implicit global configuration" {
+  set_dolt_user "baz", "baz@bash.com"
+
+  run dolt init
+  [ "$status" -eq 0 ]
+
+  run dolt config --local --get user.name
+  [ "$status" -eq 1 ]
+
+  run dolt config --local --get user.email
+  [ "$status" -eq 1 ]
+
+  assert_valid_repository
+}
+
+@test "init: explicit local configuration for name" {
+  set_dolt_user "baz", "baz@bash.com"
+
+  run dolt init --name foo
+  [ "$status" -eq 0 ]
+
+  run dolt config --local --get user.name
+  [ "$status" -eq 0 ]
+  [[ "$output" =~ "foo" ]] || false
+
+  run dolt config --local --get user.email
+  [ "$status" -eq 1 ]
+
+  assert_valid_repository
+}
+
+@test "init: explicit local configuration for email" {
+  set_dolt_user "baz", "baz@bash.com"
+
+  run dolt init --email foo@bar.com
+  [ "$status" -eq 0 ]
+
+  run dolt config --local --get user.name
+  [ "$status" -eq 1 ]
+
+  run dolt config --local --get user.email
+  [ "$status" -eq 0 ]
+  [[ "$output" =~ "foo@bar.com" ]] || false
+
+  assert_valid_repository
+}
+
+@test "init: explicit local configuration for name and email" {
+  set_dolt_user "baz", "baz@bash.com"
+
+  run dolt init --name foo --email foo@bar.com
+  [ "$status" -eq 0 ]
+
+  run dolt config --local --get user.name
+  [ "$status" -eq 0 ]
+  [[ "$output" =~ "foo" ]] || false
+
+  run dolt config --local --get user.email
+  [ "$status" -eq 0 ]
+  [[ "$output" =~ "foo@bar.com" ]] || false
+
+  assert_valid_repository
+}
+
+@test "init: no explicit or implicit configuration for name and email" {
```

Comment:
Seems like we're missing a test where `user.name` and `user.email` aren't set, but are given to `dolt init`?

Reply:
I was covering a similar case where global config was set and tested that explicit config was saved to local config, but you're right that it's worth also covering when no global config is present at all. Added now.